core/src/main/scala/org/apache/spark/sql/SnappyContext.scala

```diff
@@ -1207,15 +1219,19 @@ final class BlockAndExecutorId(private[spark] var _blockId: BlockManagerId,
  * Spark driver that also hosts a job-server and GemFireXD accessor.
  */
 case class SnappyEmbeddedMode(override val sc: SparkContext,
-    override val url: String) extends ClusterMode
+    override val url: String) extends ClusterMode {
+  override val description: String = "Embedded cluster mode"
+}
 
 /**
  * This is for the two cluster mode: one is the normal snappy cluster, and
  * this one is a separate local/Spark/Yarn/Mesos cluster fetching data from
  * the snappy cluster on demand that just remains like an external datastore.
  */
 case class SplitClusterMode(override val sc: SparkContext,
-    override val url: String) extends ClusterMode
+    override val url: String) extends ClusterMode {
+  override val description: String = "Split cluster mode"
+}
```

Comment:
Had disabled this in 0.9 for users. In the code all occurrences are still to be removed. Using SNAP-1610 to track it. Changes for SNAP-1610 are ready in another branch and will merge those after opening a PR.